servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -345,7 +345,7 @@ void RenderForwardClustered::RenderBufferDataForwardClustered::configure(RID p_c
 
 		color_msaa = RD::get_singleton()->texture_create(tf, RD::TextureView());
 
-		tf.format = RD::get_singleton()->texture_is_format_supported_for_usage(RD::DATA_FORMAT_D24_UNORM_S8_UINT, RD::TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT) ? RD::DATA_FORMAT_D24_UNORM_S8_UINT : RD::DATA_FORMAT_D32_SFLOAT_S8_UINT;
+		tf.format = RD::get_singleton()->texture_is_format_supported_for_usage(RD::DATA_FORMAT_D24_UNORM_S8_UINT, RD::TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT | RD::TEXTURE_USAGE_SAMPLING_BIT) ? RD::DATA_FORMAT_D24_UNORM_S8_UINT : RD::DATA_FORMAT_D32_SFLOAT_S8_UINT;
```

Comment:
I don't understand this change. Don't we absolutely need ``TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT``? To me this says that we can get away with using ``DATA_FORMAT_D24_UNORM_S8_UINT`` even if its unsupported as ``TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT`` 

Perhaps you could explain in more detail?

Reply:
The change means we are picking the format that, _in addition to_ being suited to a depth buffer, is also sampleable like a regular texture.

The latter is the piece I added. And it's needed since in D3D12 D24_S8 is an OK depth-stencil format but not sampleable, so we need the code to check for the full usage so it knows it has to pick D32_S8.